scvi/inference/posterior.py

```diff
@@ -273,176 +286,416 @@ def differential_expression_stats(self, M_sampling=100):
         return px_scales, all_labels
 
     @torch.no_grad()
-    def sample_scale_from_batch(self, n_samples, batchid=None, selection=None):
+    def sample_scale_from_batch(
+        self,
+        n_samples: Optional[int] = None,
+        n_samples_per_cell: Optional[int] = None,
+        batchid: Optional[Union[List[int], np.ndarray]] = None,
+        genes: Optional[Union[List[str], np.ndarray]] = None,
+        selection: Union[List[bool], np.ndarray] = None,
+    ) -> dict:
+        """
+        :param n_samples: Number of samples in total per batch (fill either `n_samples_total`
+        or `n_samples_per_cell`)
+        :param n_samples_per_cell: Number of time we sample from each observation per batch
```

Comment:
I thought you were removing `n_samples_per_cell`?

Reply:
I think the current status is acceptable. I just left an optional `n_samples_per_cell` that can replace `n_samples` just in case. Do you think I should remove it?